### PR TITLE
MGMT-14240: Plug OCI platform e2e test into CI

### PR DESCRIPTION
**Pull request by adriengentil**
Fix several fixes in order to make the OCI e2e test work in Prow:
- Fix NSGs on clusters nodes
  - Allow communications between them
  - Allow communications from the nodes to the LB 
- Prevent the creation of DNS entries on the CI machine, it's not required since we setup DNS in OCI 
- Implemented various retries, I observed a lot of flakiness from OCI
- Force user managed networking to true when platform is OCI

### CODE REVIEW

File: src/assisted_test_infra/test_infra/controllers/node_controllers/tf_controller.py [hunk 1]
```diff
@@ -101,6 +101,7 @@ def shutdown_all_nodes(self) -> None:
         for node in nodes:
             self.shutdown_node(node.name)
 
+    @abstractmethod
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/controllers/node_controllers/tf_controller.py:104]: Why annotate it with `abstractmethod`? Not sure it's needed here

Comment (adriengentil) [reply src/assisted_test_infra/test_infra/controllers/node_controllers/tf_controller.py:104]: ah ok, I thought it was required since we override it

Comment (eliorerz) [reply src/assisted_test_infra/test_infra/controllers/node_controllers/tf_controller.py:104]: Nope, `abstractmethod` raise an exception in the inheriting class if the abstract method is not implemented. It's basically force the inheriting classes to override this method

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**